### ISSUE TITLE
Some issues with Spyder projects and the workspace functionality in the PyLS

### ISSUE THREAD

**Issue by mrclary** [spyder-ide/spyder#13351]
## Problem Description

When launching Spyder with a previously open project, jedi settings are not passed to the python language server workspace. This may result in incomplete or failed command completions.

### What steps reproduce the problem?

1. launch Spyder (bootstrap) with a previously open project from an environment without a desired package (e.g. matplotlib) but have a different iPython Console environment that does have the desired package as the default python environment.
2. A minimal code file in the editor will illustrate the issue, e.g.
```python
import matplotlib.pyplot as plt
plt.
```
3. tab completions do not work
4. closing the project restores completions. Also, reopening the project restores completions.

Upon investigation I found that `pyls.workspace.Workspace._config.plugin_settings('jedi',...)` is empty until the project is closed or reopened.
I suspect that the pyls workspace is created before the jedi settings are read/configured upon launching Spyder with an open project.

This is not an issue when launching Spyder without an open project.
Restarting the LSP server has no effect.

I'm not sure where to look for the offending code...
Any help would be appreciated.

**Comment by mrclary** [comment]
I've noticed some other idiosyncrasies:
* After closing a project, the class and function selector in the editor stops working, requiring closing and reopening a document.
* When adding a split window in the editor, the class and function selector does not work in the second pane, requiring closing and reopening the document.
* The completion of supposedly preloaded modules is still slow, particularly numpy, even after a primary call. I have noticed in the lsp_logs that the pyls only preloads modules in the host environment, and never in prescribed Jedi environment. 

It seems that there may be some bug with the implementation of jedi settings with projects, and projects with split windows and the python-language-server (?)
